app/src/main/java/no/nordicsemi/android/common/test/main/page/BasicViewsPage.kt

```diff
@@ -58,7 +57,9 @@ class BasicPageViewModel @Inject constructor(
     init {
         navigator.resultFrom(Scanner)
             // Filter out results of cancelled navigation.
-            .mapNotNull { it }
+            .filter { it is NavigationResult.Success }
```

Comment:
mapNotNull { it as? NavigationResult.Success }

Reply:
I had it like that, but perhaps this way is more readable, with each step separate?

Reply:
I can see your point that it may be more readable, but I was thinking about that - you do some operation of filtering and you are loosing some information from this task in next map operator so you need to cast it again. Up to you what you want to do with this